### PR TITLE
Bump version to 4.1.0

### PR DESCRIPTION
**Pull request by szh**

### CODE REVIEW

**Review by gl-johnson — COMMENTED** [review]
LGTM - although I am not sure if the release process for this repo necessitates updating pom.mustache to reflect the new version, i.e. 
`<version>4.0.0-SNAPSHOT</version>` ---> `<version>4.1.0-SNAPSHOT</version>`

Pinging @john-odonnell or @telday

**Comment by john-odonnell** [comment]
Yeah, fairly certain we should update `pom.mustache` version before releasing.
The `bin/deploy-release` script will uses `mvn version:set -DnewVersion=${TAG}`,
which should update `client/pom.xml` to the build tag, but our template would be out of sync if it hadn't been updated.

@telday wrote all the release tooling here - he'd know for certain.

Whatever the outcome, we should definitely make note in the Contributing doc.

**Comment by telday** [comment]
@john-odonnell Is correct here, the release script does run a `mvn version:set` before it releasees which will set the correct version based on the tag you are releasing for. However the next time the `pom.xml` is regenerated it will have the old version. I actually don't think this would cause any issues, but it may be confusing if it is not kept up to date.

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,4 +37,6 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   [Google's Java Style](https://google.github.io/styleguide/javaguide.html).
   [cyberark/conjur-sdk-java#3](https://github.com/cyberark/conjur-sdk-java/issues/3)
 
-[Unreleased]: https://github.com/cyberark/conjur-openapi-spec/compare/v4.0.0...HEAD
+[Unreleased]: https://github.com/cyberark/conjur-sdk-java/compare/v4.0.0...HEAD
```
Comment (john-odonnell) [inline CHANGELOG.md:40]:
```suggestion
[Unreleased]: https://github.com/cyberark/conjur-sdk-java/compare/v4.1.0...HEAD
```

**Review by john-odonnell — CHANGES_REQUESTED** [review]
Need one small fix in the Changelog links. LGTM otherwise!

**Review by john-odonnell — APPROVED** [review]
LGTM 👍

**Pull request merged**